xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/Code/NotificationsWatchUIX.xml

```diff
@@ -0,0 +1,656 @@
+        <name>parameters</name>
+        <number>7</number>
+        <prettyName>Extension Parameters</prettyName>
+        <restricted>0</restricted>
+        <rows>10</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </parameters>
+      <scope>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>scope</name>
+        <number>8</number>
+        <prettyName>Extension Scope</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>wiki=Current Wiki|user=Current User|global=Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </scope>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <content>{{velocity}}
+#if (!$isGuest)
+#set($discard = $xwiki.ssx.use('XWiki.Notifications.Code.NotificationsWatchUIX'))
+#set($discard = $xwiki.jsx.use('XWiki.Notifications.Code.NotificationsWatchUIX'))
+#macro (_displayReferenceInModal $entityReference)
+  #if ($entityReference.type == 'WIKI')
+    &lt;span class="entity-reference"&gt;&lt;span class="fa fa-globe"&gt;&lt;/span&gt; wiki $entityReference.name&lt;/span&gt;
+  #elseif ($entityReference.type == 'SPACE')
+    &lt;span class="entity-reference"&gt;&lt;span class="fa fa-sitemap"&gt;&lt;/span&gt; space $entityReference&lt;/span&gt;
+  #elseif ($entityReference.type == 'PAGE')
+    &lt;span class="entity-reference"&gt;&lt;span class="fa fa-file"&gt;&lt;/span&gt; page $entityReference&lt;/span&gt;
+  #else
+    &lt;span class="entity-reference"&gt;$entityReference&lt;/span&gt;
+  #end
+#end
+#set ($watchedStatus = $services.notification.watch.getLocationWatchedStatus($doc))
+## If the watched status informs that the filter is about children it means it's a filter for the space,
+## so we should look for ancestor of the space.
+#set ($refForAncestorFirstFilter = $doc.getDocumentReference())
+#if ($watchedStatus == 'WATCHED_WITH_CHILDREN_FOR_ALL_EVENTS_AND_FORMATS' || $watchedStatus == 'BLOCKED_WITH_CHILDREN_FOR_ALL_EVENTS_AND_FORMATS')
+  #set ($refForAncestorFirstFilter = $doc.getDocumentReference().parent)
+#end
+#set ($ancestorFirstFilter = $services.notification.watch.getFirstFilteredAncestor($refForAncestorFirstFilter))
+#if ($ancestorFirstFilter)
+  #set ($ancestorRef = $ancestorFirstFilter.left)
+  #set ($ancestorWatchStatus = $ancestorFirstFilter.right)
+#end
+## FIXME: the different icons should use the icon service
```

Comment:
This should be fixed.

Reply:
Yes I haven't fixed this yet since I'm not yet 100% sure about the UI and I'd need to introduce new icons in the iconset for being able to use the icon service with current choice.